GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -594,8 +605,9 @@ private FileSystemTaskResult ExecuteBackgroundOperation(FileSystemTask gitUpdate
                     metadata.Add("virtualPath", gitUpdate.VirtualPath);
                     if (this.newlyCreatedFileAndFolderPaths.Contains(gitUpdate.VirtualPath))
                     {
-                        string fullPathToFolder = Path.Combine(this.context.Enlistment.WorkingDirectoryRoot, gitUpdate.VirtualPath);
-                        if (!this.context.FileSystem.FileExists(fullPathToFolder))
+                        string fullPathToFile = Path.Combine(this.context.Enlistment.WorkingDirectoryRoot, gitUpdate.VirtualPath);
+                        CheckConditionWithRetry(() => !this.context.FileSystem.FileExists(fullPathToFile), NumberOfRetriesCheckingForDeleted, MillisecondsToSleepBeforeCheckingForDeleted);

```

Comment:
Style suggestion: You could have `CheckConditionWithRetry` returned the value of `predicate()`, and that way when the `predicate()` is true you can avoid have checking it one more time:

```
bool fileDeleted = CheckConditionWithRetry(...)
if (fileDeleted)
```